### DISCUSSION TITLE
Module errors about "UFS_IC_generator.py" (CCPP-SCM v6) on Hera

### DISCUSSION THREAD

**Discussion by wzzheng90** [NCAR/ccpp-scm#495] [Cases, Initial Conditions, Forcing]
Hi,

I just cloned the latest version of CCPP-SCM on Hera and tried "UFS_IC_generator.py" script to generate the example UFS Atmosphere initial condition case input file.  I got errors when I ran it (similar to "UFS_forcing_ensemble_generator.py").

import xesmf
ModuleNotFoundError: No module named 'xesmf'

BTW, before I ran it, I did load the modules:
      % module use scm/etc/modules
      % module load hera_intel
And I also load more modules such as py-shapely/1.8.0, py-xarray/2022.3.0.   But I did not find "xesmf" on Hera.

Any suggestions? 
Thank you very much!!

Weizhong

**Comment by hertneky** [comment]
Hi Weizhong.
It sounds like you cloned the develop version of CCPP SCM, which is totally fine, as in it _should_ work. A couple things to note.
1. The scripts for generating UFS cases have been under rapid development and we are anticipating merging a PR with these changes soon for the upcoming public release v7. You could wait for this PR/release (recommended), or....
2. For running the UFS_IC_generator in the version you have cloned now, you will need to create the conda environment, which can be done using the ccpp-scm/environment-ufsreplay.yml (conda env create -f environment-ufsreplay.yml). This environment has the required python packages.

**Reply by wzzheng90** [reply]
@hertneky   Many thanks!   Yes, I cloned the develop version of CCPP SCM.    I am trying to  create the conda environment.  Under "ccpp-scm" directory, I typed "conda env create -f environment-ufsreplay.yml" but got an error such as:
       env: ‘create’: Permission denied

The python I used as follows:
%where python
/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/python-3.10.13-2p2rady/bin/python
/usr/bin/python

Any further suggestions?  Really appreciate!

Weizhong

**Reply by hertneky** [reply]
Hmmm, I wonder if it's a permissions issue? i.e. trying to create the environment on the system and you don't have root access?
I've always installed [miniconda](https://docs.anaconda.com/miniconda/) locally and then created the environment. Actually, I suggest creating the environment with mamba, as it solves the environment faster than conda:

conda install mamba -n base -c conda-forge
conda activate base
mamba env create -f environment-ufsreplay.yml
conda activate env_ufsreplay

Perhaps there are options to specify where to install the environment, but I have never tested. Maybe this would work? https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#specifying-a-location-for-an-environment
